apps/client/src/app/(user)/tables/page.tsx

```diff
@@ -1,3 +1,131 @@
-export default function page() {
-  return <div>tablesPage</div>;
+'use client';
+
+import { Calendar, Button, Pagination } from "@spin-spot/components";
+import { useTables, useTimeBlocks } from "@spin-spot/services";
+import { useAuth } from "@spin-spot/services"; 
+import { useState, useEffect } from "react";
+
+export default function Page() {
+  const [selectedDate, setSelectedDate] = useState<Date | undefined>(undefined);
+  const { data: tables } = useTables();
+  const [selectedTable, setSelectedTable] = useState<string | null>(null);
+  const { data: timeBlocks } = useTimeBlocks(selectedTable||undefined); //obtiene las timeblocks de las mesas, si no hay ninguna seleccionada, trae todas
+  const { user } = useAuth(); // Obtiene la información del usuario logeado
+
+  useEffect(() => {
+    if (tables?.length) {
+      setSelectedTable(null); // No seleccionar ninguna mesa por defecto
+    }
+  }, [tables]);
+
+  function handleReserve(timeBlockId: string) {
+    console.log(`Reserva solicitada para el bloque de tiempo con ID: ${timeBlockId}`);
+  }
+
+  function handleCancel(timeBlockId: string) {
+    console.log(`Cancelar reserva solicitada para el bloque de tiempo con ID: ${timeBlockId}`);
+  }
+
+  function handleEdit(timeBlockId: string) {
+    console.log(`Editar reserva solicitada para el bloque de tiempo con ID: ${timeBlockId}`);
+  }
+
+  function handleJoin(timeBlockId: string) {
+    console.log(`Unirse a la reserva solicitada para el bloque de tiempo con ID: ${timeBlockId}`);
+  }
+
+  const filterTimeBlocks = (blocks: any[]) => {
+    if (!selectedDate) return [];
+    return blocks.filter(block => {
+      const blockDate = new Date(block.startTime);
+      // Filtrar por fecha y, si hay una mesa seleccionada, por el código de mesa
+      const isSameDate = (
+        blockDate.getDate() === selectedDate.getDate() &&
+        blockDate.getMonth() === selectedDate.getMonth() &&
+        blockDate.getFullYear() === selectedDate.getFullYear()
+      );
+      const isSameTable = selectedTable ? block.table.code === selectedTable : true;
+      return isSameDate && isSameTable;
+    });
+  };
```

Comment:
Cambiar esto por un useMemo, para que no se recalcule en cada renderizado:

```suggestion
  const filteredTimeBlocks = useMemo<IPopulatedTimeBlock[]>(() => {
    if (!selectedDate || !timeBlocks) return [];
    return timeBlocks.filter(block => {
      const blockDate = new Date(block.startTime);
      // Filtrar por fecha y, si hay una mesa seleccionada, por el código de mesa
      const isSameDate = (
        blockDate.getDate() === selectedDate.getDate() &&
        blockDate.getMonth() === selectedDate.getMonth() &&
        blockDate.getFullYear() === selectedDate.getFullYear()
      );
      const isSameTable = selectedTable ? block.table.code === selectedTable : true;
      return isSameDate && isSameTable;
    });
  }, [timeBlocks]);
```